### DISCUSSION TITLE
Starter Example for Library Authors

### DISCUSSION THREAD

**Discussion by rwieruch** [vercel/turborepo#1109] [Ideas]
I think for library authors it would be great to have a TS starter example:

```
- apps
--- documentation/ (e.g. Next)
--- storybook/
- packages
--- @my-library/core
--- @my-library/standalone-one
--- @my-library/standalone-two
--- shared
```

Some notes:

- all components in *storybook/* get imported from the packages/library the same way as they would be imported when installing them (e.g. `import X from @my-library/standalone-one`)
- the *documentation/* is optional, but a nice add-on for creating a nicer landing page than the storybook
- each *packages/library* folder should be build and npm published on their own
- all apps/packages should use TS
- optional: it should be possible to give each package a from a base extended or a standalone TS/babel config

Over the last weeks I wanted to tackle this task myself, but didn't find the time to do it. So I just wanted to put this out for now. Maybe someone else picks it up or I will have time eventually :)

**Comment by leerob** [comment]
**Update:** We now have a design system example using:

- 🏎 [Turborepo](https://turborepo.org/) — High-performance build system for Monorepos
- 🚀 [React](https://reactjs.org/) — JavaScript library for user interfaces
- 🛠 [Tsup](https://github.com/egoist/tsup) — TypeScript bundler powered by esbuild
- 📖 [Storybook](https://storybook.js.org/) — UI component environment powered by Vite

https://github.com/vercel/turborepo/tree/main/examples/design-system

**Reply by rwieruch** [reply]
This is awesome. Thanks for putting this together @leerob 🚀

**Reply by rwieruch** [reply]
Maybe worth mentioning: I tried using with pnpm workspaces instead of yarn and got ` Error: [vite-plugin-mdx] "@mdx-js/react" must be installed`. Seems to be some issue with pnpm + Vite. However, rolling back to yarn worked again.
